.github/workflows/update-changelog.yaml

```
@@ -0,0 +1,96 @@
+name: Update changelog
+
+concurrency:
+  group: update-changelog-${{ github.event_name }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        type: string
+        required: true
+        description: 'EMQX version (with v/e prefix)'
+      repository:
+        type: choice
+        required: true
+        options:
+        - emqx/emqx
+        - emqx/emqx-platform
+      run_id:
+        type: string
+        required: true
+        description: 'The run id of the workflow that generates the changelog'
+
+permissions:
+  contents: read
+
+jobs:
+  update-changelog:
+    runs-on: ubuntu-latest
+    env:
+      EMQX_VERSION: ${{ github.event.inputs.version }}
+      REPOSITORY: ${{ github.event.inputs.repository }}
+      RUN_ID: ${{ github.event.inputs.run_id }}
+    permissions:
+      contents: write
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Generate GitHub App token
+        id: app-token
+        uses: actions/create-github-app-token@v1
+        with:
+          app-id: ${{ vars.AUTH_APP_ID }}
+          private-key: ${{ secrets.AUTH_APP_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+
+      - uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: changes
+          repository: ${{ env.REPOSITORY }}
+          run-id: ${{ env.RUN_ID }}
+          github-token: ${{ steps.app-token.outputs.token }}
+
+      - name: create release branch and PR
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -euxo pipefail
+          RE='^[ve]([0-9]+)\.([0-9]+)\.([0-9]+)(-([a-z]+\.[0-9]+))?$'
```

Should it be:

```suggestion
          RE='^[ve]([0-9]+)\.([0-9]+)\.([0-9]+)(-([A-Za-z0-9.]+))?$'
```

?

---

we never use capital letters in tag modificators like `-alpha`, `-beta` and `-rc`.

---

This would work too, however, we never use capital letters in tag modificators `-alpha`, `-beta` and `-rc`, and the current version is more strict.